containers/Labels/modals/Edit.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { c } from 'ttag';
+import { FormModal, useEventManager, useNotifications, useApiWithoutResult } from 'react-components';
+import { LABEL_TYPES, LABEL_COLORS } from 'proton-shared/lib/constants';
+import { randomIntFromInterval } from 'proton-shared/lib/helpers/function';
+import { createLabel, updateLabel } from 'proton-shared/lib/api/labels';
+import { noop } from 'proton-shared/lib/helpers/function';
+
+import NewLabelForm from '../../../components/Labels/NewLabelForm';
+
+function EditLabelModal({ label, mode, onEdit, onAdd, ...props }) {
+    const { call } = useEventManager();
+    const { createNotification } = useNotifications();
+    const reqCreate = useApiWithoutResult(createLabel);
+    const reqUpdate = useApiWithoutResult(updateLabel);
+
+    const I18N = {
+        edition({ Name, Exclusive } = {}) {
+            if (Exclusive === LABEL_TYPES.LABEL) {
+                return c('Label/folder modal').t`Edit label: ${Name}`;
+            }
+            return c('Label/folder modal').t`Edit folder: ${Name}`;
+        },
+        create({ Name } = {}, type) {
+            if (type === 'label') {
+                return c('Label/folder modal').t`Create a new label: ${Name}`;
+            }
+            return c('Label/folder modal').t`Create a new folder: ${Name}`;
+        }
+    };
+
+    const [model, setModel] = useState(
+        label || {
+            Name: '',
+            Color: LABEL_COLORS[randomIntFromInterval(0, LABEL_COLORS.length - 1)],
+            Exclusive: +(props.type === 'folder')
+        }
+    );
+
+    useEffect(() => {
```

Comment:
Why do you need to set the `model` when the model change? It seems useless right?